### PR TITLE
Upgrade dependencies, docs and remove unused imports

### PR DESCRIPTION
**Pull request by mmoayyed**
This pull request:

1. Clarifies the project JDK requirement (JDK 11 cannot be used to build)
2. Upgrade a number of dependencies including SB
3. Sets the scope for lombok to be `compile` so as to not have it *leak* in the final published pom.
4. Removes all unused imports.

### CODE REVIEW

**Comment by mmoayyed** [comment]
To clarify, tests under JDK11 fail as such:

```
Caused by: java.lang.IllegalStateException: Error during attachment using: net.bytebuddy.agent.ByteBuddyAgent$AttachmentProvider$Compound@be0bf9ee
Caused by: java.lang.reflect.InvocationTargetException
Caused by: java.io.IOException: Can not attach to current VM

[ERROR] exchangeAuthorizationCodeFor2FAResult_network_failure  Time elapsed: 0 s  <<< ERROR!
java.lang.IllegalStateException: Could not initialize plugin: interface org.mockito.plugins.MockMaker
	at com.duosecurity.service.DuoConnectorTest.exchangeAuthorizationCodeFor2FAResult_network_failure(DuoConnectorTest.java:87)
Caused by: java.lang.IllegalStateException: Failed to load interface org.mockito.plugins.MockMaker implementation declared in java.lang.CompoundEnumeration@5023bb8b
Caused by: org.mockito.exceptions.base.MockitoInitializationException: 

Could not initialize inline Byte Buddy mock maker. (This mock maker is not supported on Android.)
```

File: duo-universal-sdk/pom.xml [hunk 1]
```diff
@@ -115,12 +115,13 @@
         <dependency>
             <groupId>org.projectlombok</groupId>
             <artifactId>lombok</artifactId>
-            <version>1.18.10</version>
+            <version>1.18.16</version>
+            <scope>compile</scope>
```
Comment (jeffreyparker) [inline duo-universal-sdk/pom.xml:119]: Thank you for pointing out this potential optimization! I _believe_ we actually want the `provided` scope here, as my understanding is that `compile` is the default and wouldn't actually change the behavior. https://stackoverflow.com/questions/29385921/maven-scope-for-lombok-compile-vs-provided

**Review by jeffreyparker — COMMENTED** [review]
Thank you for the great (and quick!) PR! Just one small comment, but overall this is a great cleanup!

File: duo-universal-sdk/pom.xml (hunk 1 above)
Comment (mmoayyed) [reply duo-universal-sdk/pom.xml]: Absolutely; updated.

By the way, I didn't go that far but you should also take a look at the project hierarchy and the parent tags for the sub-modules. Maven's quite forgiving, but I am not so sure if the current structure is entirely accurate. Parent references seem to be missing or are misplaced.

**Comment by vbscott** [comment]
Thank you for your PR.  Unfortunately the changes to [duo-example/pom.xml](https://github.com/duosecurity/duo_universal_java/pull/3/files#diff-36357aa13d4bfc3473ea066b207da83503120ac23f74ba14900a7036bd62e573R8) cause problems with duo-example.  We can't directly merge in this diff, due to our internal repository configuration, but I can add the rest of your changes in.  We can also look into the duo-example/pom.xml changes in a future PR/issue.

**Comment by mmoayyed** [comment]
No worries, thanks for the review and update. I am curious; did you cherry-pick commits from this PR when you merged? Not an issue at all in any way, but I would have thought that my name and commit/changes would show up in the commit log here?

**Comment by mmoayyed** [comment]
Ah never mind; sorry. I see how that might not have been possible. Thanks again!

**Pull request closed without merging**